add xdg support

legrostdg opened the pull request:
Should fix #771

For now I've just put the pronsole history file in user_data_dir/pronsole/history if ~/.pronsole-history does not exist. Let me know if you agree this, and I'll add user_config_dir support for pronsolerc.

Please note that win32 and macos are supported by appdirs.

rockstorm101 commented:
Hi @legrostdg, I'm sorry it took me this long to review this. By the way its written it should not break any current installation. I've tested on Debian, but I cannot speak for Windows and Mac OS.

Anyway, if you don't mind me asking, I've never used appdirs. According to the [source], `user_data_dir` is defined as:
```
def user_data_dir(appname=None, appauthor=None, version=None, roaming=False)
```
Yet you invoke it as:
```
user_data_dir("Printrun", "pronsole")
```
Why the second argument? Why "_pronsole_"?

Thanks

[source]: https://github.com/ActiveState/appdirs/blob/master/appdirs.py

legrostdg commented:
@rockstorm101 I made a typo, I wanted to use "pronsole" as appname, and thought appauthor was kind of necessary for Windows. According to the source, appauthor falls back to appname, so we can just set appname. I don't know if you prefer "pronsole", "printrun" or "Printrun" there.

rockstorm101 commented:
Hi @legrostdg, thanks for answering. I would go with "Printrun" as the _appname_ but I'll leave that decision to the project owners. (CC @kliment @iXce)

kliment commented:
Agreed on "Printrun"

On 07.01.2018 12:27, Rock Storm wrote:
> Hi @legrostdg <https://github.com/legrostdg>, thanks for answering. I
> would go with "Printrun" as the /appname/ but I'll leave that decision
> to the project owners. (CC @kliment <https://github.com/kliment> @iXce
> <https://github.com/ixce>)
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kliment/Printrun/pull/866#issuecomment-355815701>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAN7c1_Drt7FQpfmOMDfV1Q7W8BSGyqdks5tIKoOgaJpZM4Q_bLh>.
>

legrostdg commented:
OK, I've updated the appname to "Printrun" and added support for supporting user_config_dir. I was not sure of what to do with the sys.frozen logic on Windows, so I just removed this.

kliment commented:
The sys.frozen logic needs to stay - otherwise people who used printrun with a packaged executable will lose their config when they switch to this version. It was originally introduced because files that start with a . confuse some versions of windows explorer

legrostdg commented:
@kliment I reintroduced the sys.frozen logic, can you have a look to this?

legrostdg commented:
I still need to put the ~bak file in user_cache_dir/Printrun/.

legrostdg commented:
done.

rockstorm101 commented:
Thanks a lot for your work. I really like this feature. If @kliment does not beat me to it I'll merge as soon as I finish testing.

kliment commented:
Looks good to me, @rockstorm101

legrostdg commented:
Thank you for the review! I hope to see 1.6.1 soon :)

rockstorm101 commented:
Merging. Thanks again.
